runtime/include/chpl-cache.h

```diff
@@ -36,7 +36,7 @@ int chpl_cache_enabled(void)
 {
   // The remote cache uses thread local storage, so if tasks can migrate
   // between threads we lose out ability to correctly fence.
-  return CHPL_CACHE_REMOTE && !chpl_task_canMigrateThreads();
+  return CHPL_CACHE_REMOTE && chpl_numNodes > 1 && !chpl_task_canMigrateThreads();
```

Comment:
The cost of the extra load is almost certainly too small to ever notice, and I don't think this is ever called from performance critical code. What perf-critical functions did you see?

I added this because it seems weird to me that we never call `chpl_cache_comm_{put,get}` under `-nl 1`, but `chpl_cache_fence` was running.

Reply:
That said, I don't feel strongly about this. If you care, I'll take this out but I do not expect it to have any performance implications.

Reply:
> I added this because it seems weird to me that we never call chpl_cache_comm_{put,get} under -nl 1, but chpl_cache_fence was running.

That does seem weird and just wasn't clear to me from the PR message.

I'm OK with the extra load for the moment, especially if you keep an eye on the performance.

> What perf-critical functions did you see?

I've seen in the past code that calls some of the comms functions at an unreasonable rate, but didn't have a specific example. (In particular AFAIK the main comm_gen_get functions will be OK since these already have a check for if local and will take more time if remote).

Reply:
Updated PR message.

I'll keep this in mind when looking at remote-cache perf.

Reply:
Oh, I bet there are some benchmarks that call `chpl_cache_fence` a zillion times in a row...

Reply:
That seems likely, but `chpl_cache_fence()` is a non-inlined function that accesses task local storage, this is not going to add noticeable overhead.

Reply:
Ah, thread local, not task local. Accessing thread local should be free, but task-local has some overhead currently.

Reply:
Discussed offline, we decided to drop it for now out of paranoia